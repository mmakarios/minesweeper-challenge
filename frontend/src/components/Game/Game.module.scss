.game {
}

.afterGameActions {
  margin-top: 5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
