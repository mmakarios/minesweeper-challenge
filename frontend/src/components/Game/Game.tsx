import { useCallback } from "react";
import styles from "./Game.module.scss";
import { BoardContextType, Board } from "../../types";
import { useEffect, useState, createContext } from "react";
import { retrieveBoard } from "../../apis";
import { useParams } from "react-router-dom";
import GameBoard from "../GameBoard";

export const BoardContext = createContext<BoardContextType>({});

export const Game = () => {
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [alert, setAlert] = useState<string>("");
  const [board, setBoard] = useState<Board>();
  const { id } = useParams<{ id: string }>();

  useEffect(() => {
    async function fetchBoard() {
      console.log(`fetching board`);
      setIsLoading(true);
      setAlert("");
      let board;
      try {
        board = await retrieveBoard(id);
      } catch (err) {
        setAlert("Error fetching the board!");
      }
      console.log("fetch board: ", board);
      setBoard(board);
      setIsLoading(false);
    }

    fetchBoard();
  }, [id]);

  const onBoardChange = useCallback(
    (board: Board) => {
      setBoard(board);
    },
    [setBoard]
  );

  return (
    <div className={styles.Game}>
      {isLoading && <div>Loading game...</div>}
      <BoardContext.Provider value={{ board, onBoardChange }}>
        <GameBoard />
      </BoardContext.Provider>
    </div>
  );
};

export default Game;
