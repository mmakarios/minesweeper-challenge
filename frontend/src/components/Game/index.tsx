export { default, BoardContext } from "./Game";
