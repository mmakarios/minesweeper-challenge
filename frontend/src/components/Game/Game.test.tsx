import React from "react";
import renderer from "react-test-renderer";
import Game from "./Game";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    environment: "dev",
    service: "fakeService",
    id: "1",
  }),
}));

it("renders component", () => {
  const tree = renderer.create(<Game />).toJSON();

  expect(tree).toMatchSnapshot();
});
