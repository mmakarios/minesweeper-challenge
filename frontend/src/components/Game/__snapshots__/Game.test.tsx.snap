// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div
  className="Game"
>
  <div
    className="gameboard"
  />
  
  <p>
    Loading game...
  </p>
</div>
`;
